hydra/hydra.py

```diff
@@ -327,26 +328,29 @@ def get_case_comments(self, case_number):
         return comment_data
 
     def get_case_associates(self, case_number):
-        data = []
 
-        ## The data returned from users is too much (filtered return)
+        # The data returned from users is too much (filtered return)
         def get_case_user_details(associate):
             user = self.__get_api("users/{}".format(associate["OwnerId"]))
-            data.append(
-                {
-                    "role": associate["role"],
-                    "name": user["fullName"],
-                    "title": user["fullTitle"],
-                    "irc_nick": user["ircNick"],
-                    "ooo": user["outOfOffice"],
-                    "phone": user["phone"],
-                    "email": user["email"],
-                    "region": user["superRegion"],
-                }
-            )
+            return {
+                "role": associate.get("role"),
+                "name": user.get("fullName"),
+                "title": user.get("fullTitle"),
+                "irc_nick": user.get("ircNick"),
+                "ooo": user.get("outOfOffice"),
+                "phone": user.get("phone"),
+                "email": user.get("email"),
+                "region": user.get("superRegion")
+            }
 
         associates = self.__get_api("cases/{}/associates".format(case_number))
-        self.thread_pool.map(get_case_user_details, associates, chunksize=1)
+
+        data = []
+        with ThreadPoolExecutor() as thread_executor:
```

Comment:
Moving discussion here so it can be threaded.
> Do we need the threading in the module? I'm all for it but shouldn't that be handled client side?

I think for the most part, yes, we should leave threading to client side, but A) Dave didn't ADD threading here, just reworked it and B) I think this is a different thing to thread.

To compare, I think threading here makes sense as we are getting _all_ of the attached associates from a single resource (one case) vs. the threading necessary to get _all_ of the results for a case search where each case is a different resource. 

Idk...I am open for discussion on this, but I currently lean towards this sort of threading being okay/making sense.
@sferich888 @davemulford thoughts?

Reply:
Threading already existed with the `self.thread_pool.map()` call that I replaced.

I replaced it, because it was showing issues where the `get_case_associates` function was exiting before all of the inner functions were completing, which caused an issue with the `self.thread_pool` instance when the `hydra_api.__del__` function was eventually run. It felt like a strange race condition, and my debugging efforts weren't coming up with anything useful.

Reply:
This is a good explanation to me. @sferich888 I'd like you to weigh in here so we can finish this PR before we move to the other one.

Reply:
Yea works for me, just wanted documentation of why we swapped it out.